### PR TITLE
mwan3: code cleanup and fixes

### LINKED ISSUE #13823: mwan3: can't determine gateway of openvpn tap interface
OpenWrt 19.07.4, r11208-ce6496d796

interface information is empty for openvpn tap device

```
root@OpenWrt:~# ubus call network.interface dump

                {
                        "interface": "vpn",
                        "up": true,
                        "pending": false,
                        "available": true,
                        "autostart": true,
                        "dynamic": false,
                        "uptime": 503,
                        "l3_device": "tap0",
                        "proto": "none",
                        "device": "tap0",
                        "metric": 0,
                        "dns_metric": 0,
                        "delegation": true,
                        "ipv4-address": [

                        ],
                        "ipv6-address": [

                        ],
                        "ipv6-prefix": [

                        ],
                        "ipv6-prefix-assignment": [

                        ],
                        "route": [

                        ],
                        "dns-server": [

                        ],
                        "dns-search": [

                        ],
                        "neighbors": [

                        ],
                        "inactive": {
                                "ipv4-address": [

                                ],
                                "ipv6-address": [

                                ],
                                "route": [

                                ],
                                "dns-server": [

                                ],
                                "dns-search": [

                                ],
                                "neighbors": [

                                ]
                        },
                        "data": {

                        }
                },
```

Here is my openvpn configure to set the gateway:

```
route-nopull
route 0.0.0.0 0.0.0.0 vpn_gateway 30
```

My tap device
```
369: tap0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UNKNOWN group default qlen 100
    link/ether xx:xx:xx:xx:xx:xx brd ff:ff:ff:ff:ff:ff
    inet xxx.xxx.xxx.xxx/24 brd xxx.xxx.xxx.255 scope global tap0
       valid_lft forever preferred_lft forever
    inet6 xxxx/64 scope global 
       valid_lft forever preferred_lft forever
    inet6 xxxx/64 scope link 
       valid_lft forever preferred_lft forever
```

And route table

```
default via xxx.xxx.xxx.1 dev tap0 metric 30 
xxx.xxx.xxx.0/24 dev tap0 proto kernel scope link src xxx.xxx.xxx.xxx
```

It maybe hard to get the gateway of the tap device, is there any alternative way to manually set or override the information for mwan3 ?

### LINKED ISSUE #13865: mwan3: load balancing displayed offline in multi-dial state
image

### PR DESCRIPTION
**Pull request by feckert**
Maintainer: me
Compile tested: x86_64, APU3, OpenWrt master
Run tested: x86_64, APU3, OpenWrt master

Description:
This is a maintenance P/R with the following goals.
- Clean up the source code
- Fix pending errors through community tests which are found through different test scenarios

Fixes #13823 
Fixes #13865

### CODE REVIEW

**Comment by aaronjg** [comment]
I'm curious what the motivation is on: 85845b244c171dad16e2f1c8193a06d602dba8c6

When testing, I generally just rsync the files directory over to my router to test changes, and removing the executable flag on the shell scripts will make this a bit harder. 

Thanks for all the cleanup!

**Comment by feckert** [comment]
> When testing, I generally just rsync the files directory over to my router to test changes, and removing the executable flag on the shell scripts will make this a bit harder.

You can still do the sync and I haven't changed the structure. Personally, I think it's better if I explicitly specify in the Makefile what is installed. Then I can see that at once.

I currently have a led handler for the mwan3 in development that I also want to install into the mwan3 package as a separate package that I don't always want to install.

**Comment by aaronjg** [comment]
> > When testing, I generally just rsync the files directory over to my router to test changes, and removing the executable flag on the shell scripts will make this a bit harder.
> 
> You can still do the sync and I haven't changed the structure. Personally, I think it's better if I explicitly specify in the Makefile what is installed. Then I can see that at once.

That part is fine. I was wondering about the permission change to 644 from 755.

> I currently have a led handler for the mwan3 in development that I also want to install into the mwan3 package as a separate package that I don't always want to install.

Very cool.

File: net/mwan3/files/usr/sbin/mwan3track [hunk 1]
```diff
@@ -169,6 +166,10 @@ main() {
 	INTERFACE=$1
 	STATUS=""
 	STARTED=0
+	TRACK_OUTPUT=$MWAN3TRACK_STATUS_DIR/$INTERFACE/TRACK_OUTPUT
```
Comment (aaronjg) [inline net/mwan3/files/usr/sbin/mwan3track:169]: :+1: This is an important fix. ty

**Comment by wulfy23** [comment]
ib cosmetics if you can roll into this would be super!

```
#################################################
Finalizing root filesystem...
/ABC/build_dir/target-aarch64_cortex-a72_musl/root-bcm27xx/etc/init.d/mwan3: line 3: /lib/functions.sh: No such file or directory
/ABC/build_dir/target-aarch64_cortex-a72_musl/root-bcm27xx/etc/init.d/mwan3: line 4: /lib/mwan3/common.sh: No such file or directory
/ABC/build_dir/target-aarch64_cortex-a72_musl/root-bcm27xx/etc/init.d/mwan3: line 5: /lib/functions/network.sh: No such file or directory
/ABC/build_dir/target-aarch64_cortex-a72_musl/root-bcm27xx/etc/init.d/mwan3: line 6: /lib/mwan3/mwan3.sh: No such file or directory
Create '/fs/sdd1/openwrt/RTNGext/ibminiSINGLE/r14869-rngNprocdbump-vimfull/build_dir/target-aarch64_cortex-a72_musl/root-bcm27xx/etc/filesystems'.
Add 'ntfs-3g' to known filesystems
################################################
./etc/init.d/mwan3: line 3: /lib/functions.sh: No such file or directory
./etc/init.d/mwan3: line 4: /lib/mwan3/common.sh: No such file or directory
./etc/init.d/mwan3: line 5: /lib/functions/network.sh: No such file or directory
./etc/init.d/mwan3: line 6: /lib/mwan3/mwan3.sh: No such file or directory
``

**Comment by aaronjg** [comment]
Can we merge https://github.com/openwrt/packages/pull/13853/commits/81e6a8fd3be57cba3b5bcbeb40fbbef52779f70b and https://github.com/openwrt/packages/pull/13853/commits/94a25dbd92adcf070e10a508f5103d6914449f2a ASAP?  There were some bugs that I introduced introduced in 2.10.1 that are causing people issues in the snapshot builds, and these two commits will help fix them.

**Comment by feckert** [comment]
@aaronjg 
I did a compile and run test on my system this weekend.
So I have merged the whole PR and not only the addressed commits https://github.com/openwrt/packages/commit/94a25dbd92adcf070e10a508f5103d6914449f2a and https://github.com/openwrt/packages/commit/81e6a8fd3be57cba3b5bcbeb40fbbef52779f70b

I think we need another P/R soon.

**Comment by aaronjg** [comment]
Okay, thanks. That takes care of the major bugs that were introduced in my last set of changes.  I'm now getting some errors regarding `ipset` when starting mwan3. I'll stage a new P/R to fix.

**Pull request merged**